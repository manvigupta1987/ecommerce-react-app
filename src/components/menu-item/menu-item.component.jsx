import React from 'react';
import {withRouter} from 'react-router-dom'; // withRouter is the HOC component. 

import './menu-item.styles.scss'
import {MenuItemContainer,ContentContainer,
        BackgroundImageContainer, 
        TitleContainer,
        ContentSubtitle } from './menu-item.styles.jsx'

export const MenuItem = ({section, history, match}) => {
  return (
    <MenuItemContainer size={section.size} onClick={()=>history.push(`${section.linkUrl}`)}>
      { /* <div style = {{ backgroundImage: `url(${section.imageUrl})`}} className='background-image'></div> */}
      <BackgroundImageContainer className='background-image' imageUrl={section.imageUrl}/>
      <ContentContainer className='content'>
        <TitleContainer>{section.title.toUpperCase()}</TitleContainer>
        <ContentSubtitle>SHOP NOW</ContentSubtitle>
      </ContentContainer>
    </MenuItemContainer>
  );
}

export default withRouter(MenuItem) ; //WithRouter will now return us the access to history, location, match props. 