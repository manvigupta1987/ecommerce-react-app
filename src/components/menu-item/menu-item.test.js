
import React from 'react';
import { shallow } from 'enzyme';
import {MenuItem} from './menu-item.component';

describe('MenuItem component testing', () => {
  let mockSection = {
    size: 'large',
    linkUrl: '/hats',
    imageUrl: 'testImage',
    title: 'testing'
  }
  let wrapper;
  let mockHistory;

  beforeEach(()=>{
    mockHistory = {
      push: jest.fn()
    }

    const mockProps = {
      section: mockSection,
      history: mockHistory
    }

    wrapper = shallow(<MenuItem {...mockProps} />)
  });

  it('MenuItem Snapshot testing', ()=> {
    expect(wrapper).toMatchSnapshot();
  })

  it('History push should be called when onClick event happens', ()=> {
    wrapper.find('MenuItemContainer').simulate('click')
    expect(mockHistory.push).toHaveBeenCalledWith(mockSection.linkUrl)
  })

  it('It should pass the value of size to the MenuItemContainer', ()=> {
    expect(wrapper.find('MenuItemContainer').prop('size')).toBe(mockSection.size);
  })

  it('Background image should have a imageUrl prop', ()=> {
    expect(wrapper.find('BackgroundImageContainer').prop('imageUrl')).toBe(mockSection.imageUrl);
  })

  it('expect title container to have some text', ()=> {
    expect(wrapper.find('TitleContainer').text()).toEqual(mockSection.title.toUpperCase());
  })
})