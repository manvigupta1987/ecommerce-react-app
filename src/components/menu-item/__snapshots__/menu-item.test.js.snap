// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component testing MenuItem Snapshot testing 1`] = `
<MenuItemContainer
  onClick={[Function]}
  size="large"
>
  <BackgroundImageContainer
    className="background-image"
    imageUrl="testImage"
  />
  <styled.div
    className="content"
  >
    <TitleContainer>
      TESTING
    </TitleContainer>
    <styled.span>
      SHOP NOW
    </styled.span>
  </styled.div>
</MenuItemContainer>
`;
